### PR TITLE
#184637142 feat: User update password

### PR DESCRIPTION
**Pull request by mjeanbosco19**
### What does this PR do?

- User should be able to update their password

### Description of Task to be completed?

- [x] The user might want to change the password in case they want to. this is different from resetting the password they have forgot.

1. GIVEN user is validly registered and account is validated.

2. THEN users new password should be hashed and updated, or rejected due to the user not remembering old password or new password and confirm password fields not matching. 

### How should this be manually tested?

**Installation**
**1. Clone the repository by running** 
```
git clone https://github.com/atlp-rwanda/destructors-ec-bn.git
```
**2. Navigate to the project directory by running cd destructors-ec-bn.**
```
cd destructors-ec-bn
```
```
cd ft-update-password-184637142
```
**3. Install the necessary dependencies by running npm install.**
```
npm install
```
**4. Run the development server by running npm run dev. This will start the server using Nodemon, which will automatically reload the server when changes are made to the code.**
```
npm run dev
```
**5. Before update password roles login and save token in headers** 
```
POST: /api/v1/users/login
```
**6. Update your password on below endpoint** 
```
PATCH: /api/v1/users/update-password'
```
**7. Run the tests by running npm run test** 
```
npm run test
```
### Any background context you want to provide?
when running this on local server put into consideration the below information 👇 

- [x] add the your `login token` in headers with a key of `Authorization` 

### What are the relevant pivotal tracker/Trello stories?

- Tracker: 184637142

### Screenshots (if appropriate)

- none

### Questions:

- none

### CODE REVIEW

File: __test__/update-password.test.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
+const request = require('supertest');
+import app, { connectDB } from '../src/app';
+const User = require('../src/models/user');
+
+connectDB(); // connect database
+
+describe('Update password feature', () => {
+  let user;
+  let token;
```
Comment (Abdulkeza) [inline __test__/update-password.test.js]: you can use only one line   => `let user, token`

File: src/middlewares/authMiddleware.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  return (req, res, next) => {
+    if (!roles.includes(req.user.role)) {
+      return res.status(403).json({
+        status: 'fail',
+        message: 'You do not have permission to perform this action.',
+      });
+    }
+    next();
+  };
+};
```
Comment (Abdulkeza) [inline src/middlewares/authMiddleware.js]: is this yours?

File: src/routes/index.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import express from 'express';
 import user from './api/user.routes.js';
 import product from './api/products.routes.js';
+import update from './api/updatePassword.routes';
```
Comment (Abdulkeza) [inline src/routes/index.js]: Please what do you mean by **update**? please use good name

**Review by Abdulkeza — CHANGES_REQUESTED** [review]
left feedback bellow

File: __test__/update-password.test.js (hunk 1 above)
Comment (mjeanbosco19) [reply __test__/update-password.test.js]: > you can use only one line => `let user, token`

Fixed

File: src/middlewares/authMiddleware.js (hunk 2 above)
Comment (mjeanbosco19) [reply src/middlewares/authMiddleware.js]: > is this yours?

Removed and use middleware from develop

File: src/routes/index.js (hunk 3 above)
Comment (mjeanbosco19) [reply src/routes/index.js]: > Please what do you mean by **update**? please use good name

Fixed, thanks

**Pull request closed without merging**